File: ruby/get_weather.rb
require 'open-uri'
require 'json'
require 'pry'

def get_weather(lat, long)
  url = "https://api.weather.gov/points/#{lat},#{long}"
  puts "Fetching Weather for location #{lat}, #{long}"
  uri = URI.open(url).read
  data = JSON.parse(uri)
  forecast = data['properties']['forecast']
  forecast_uri = URI.open(forecast).read
  forecast_data = JSON.parse(forecast_uri)
  today = forecast_data['properties']['periods'][0]
  weather = today['detailedForecast']
  weather
end

File: python/get_weather.py
import requests
# Pull weather data based on Longitude and Latitude from XML. From response pull endpoint for 12 hour forecasts for the next 7 days. From this response pull today's weather as string value

def get_weather(lat, long):
    url =  f'https://api.weather.gov/points/{lat},{long}'
    print(f'Fetching Weather for location {lat}, {long}')
    req = requests.get(url)
    assert req.status_code == 200,f"Unable to get weather data: {req.status_code}"
    resp = req.json()
    forecast = resp['properties']['forecast']
    fetch_forecast = requests.get(forecast).json()
    today = fetch_forecast['properties'].get('periods')[0]
    weather = today.get('detailedForecast')
    return weather